posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -567,29 +567,52 @@ def _convert_to_decimal_or_none(x: decimal.Decimal | float | None) -> decimal.De
 
                 return decimal.Decimal(str(x))
 
+            def _convert_to_float_or_none(x: float | None) -> float | None:
+                if x is None:
+                    return None
+
+                if math.isnan(x) or np.isinf(x):
+                    return None
+
+                return x
+
             all_values = columnar_table_data[field_name].tolist()
```

Comment:
thought: Loading all of this up is probably very memory intensive, would be nice if we could just pass Arrow tables all through the pipeline.

Reply:
Yep, although the max size of a single item here is a Decimal256 (32 bytes). We cap the rows in a table at 20,000 and so we're looking at 625 KB to load this into memory - so we should be all good 🙏